components/ledger/internal/machine/vm/machine.go

```diff
@@ -607,16 +611,19 @@ func (m *Machine) ResolveResources(ctx context.Context, store Store) ([]string,
 		m.Resources = append(m.Resources, val)
 	}
 
-	involvedAccounts := make([]string, 0)
-	involvedSources := make([]string, 0)
-	for _, accountAddress := range involvedAccountsMap {
```

Comment:
⚠️ **IMPORTANT DIFF**
Instead of marking every possible account that appears within the program (directly or indirectly - e.g. the result of `meta()`) as readLock account, we only iterate through `m.Program.ReadLockAccounts`